support/util/expose.go

```diff
@@ -42,3 +42,16 @@ func UseDedicatedDNSForKASByHC(hc *hyperv1.HostedCluster) bool {
 		// and later is used to annotate the route so the external DNS controller can watch it.
 		apiServerService.Route != nil && apiServerService.Route.Hostname != ""
 }
+
+func ServiceExternalDNSHostname(hcp *hyperv1.HostedControlPlane, serviceType hyperv1.ServiceType) string {
+	service := ServicePublishingStrategyByTypeForHCP(hcp, serviceType)
+	if service != nil && IsPublicHCP(hcp) {
+		if service.Type == hyperv1.LoadBalancer && service.LoadBalancer != nil {
+			return service.LoadBalancer.Hostname
+		} else if service.Type == hyperv1.Route && service.Route != nil {
```

Comment:
nit: for the sake of readability I'd avoid else + nested if. This could be:

```
if service.Type == hyperv1.LoadBalancer && service.LoadBalancer != nil {
    return service.LoadBalancer.Hostname
}

if service.Type == hyperv1.Route && service.Route != nil {
    return service.Route.Hostname
}
```